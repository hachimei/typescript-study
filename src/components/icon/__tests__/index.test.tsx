import React from 'react';
import { render, configure } from 'enzyme';
import Icon from "..";
import Adapter from 'enzyme-adapter-react-16';
import  renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


describe('Icon', () => {
  it('should render to a <i class="xxx"></i>', () => {
    const wrapper = render(
      <Icon type="appstore" className="my-icon-classname" />
    );
    expect(wrapper).toMatchSnapshot();
});

  it('should render correctly', () => {
    const wrapper2 = renderer.create(<Icon type="appstore" className="my-icon-classname" />)
    .toJSON();
    expect(wrapper2).toMatchSnapshot();
  });
});