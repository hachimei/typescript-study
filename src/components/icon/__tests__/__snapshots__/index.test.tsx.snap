// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<i
  className="anticon anticon-appstore my-icon-classname"
/>
`;

exports[`Icon should render to a <i class="xxx"></i> 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "anticon anticon-appstore my-icon-classname",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
